### PR TITLE
[Merged by Bors] - feat(Combinatorics/SimpleGraph): Add definition of circulant graph and cycle graph

### PR DESCRIPTION
**Pull request by IvanRenison**
Co-authored-by: Bhavik Mehta <bhavikmehta8@gmail.com>

---
This pr replaces #14364

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Circulant.lean [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  exact Or.symm h
+
+instance [DecidableEq G] [DecidablePred (· ∈ s)] : DecidableRel (CirculantGraph s).Adj :=
+  fun _ _ => inferInstanceAs (Decidable (_ ∧ _))
+
+theorem circulantGraph_adj_translate {s : Set G} {u v d : G} :
+    (CirculantGraph s).Adj (u + d) (v + d) ↔ (CirculantGraph s).Adj u v := by simp
+
+/-- Cycle graph over `Fin n` -/
+def cycleGraph : (n : ℕ) → SimpleGraph (Fin n)
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Circulant.lean:59]:
```suggestion
def CycleGraph : (n : ℕ) → SimpleGraph (Fin n)
```

File: Mathlib/Combinatorics/SimpleGraph/Circulant.lean [hunk 2]
```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2024 Iván Renison, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison, Bhavik Mehta
+-/
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+import Mathlib.Data.Set.Pointwise.Basic
+
+/-!
+# Definition of circulants graphs
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Circulant.lean:10]:
```suggestion
# Definition of circulant graphs
```
and similar below

**Review by b-mehta — COMMENTED** [review]
I think there's lots we could add about these graphs, but this is definitely a great start, and further things can be added in later PRs.
I've written too much of this to merge it however, so I'll ask another maintainer to take a look too.

maintainer merge

File: Mathlib/Combinatorics/SimpleGraph/Circulant.lean (hunk 1 above)
Comment (IvanRenison) [reply Mathlib/Combinatorics/SimpleGraph/Circulant.lean:59]: I used `cycleGraph` for using the same style that `pathGraph`, you think that `CycleGraph` is better?

Comment (IvanRenison) [reply Mathlib/Combinatorics/SimpleGraph/Circulant.lean:59]: Also, `completeGraph`, `emptyGraph`, `completeBipartiteGraph`, `hasse` are in lower case, so I will change `CirculantGraph`

**Comment by b-mehta** [comment]
good point, my mistake

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**